### PR TITLE
Asking for more materials than there is just dumps all materials now

### LINKED ISSUE #6956: Attempting to withdraw more sheets from the ore redemption machine than are available should just give you all the sheets.
### Description
Right now, the ore redemption machine doesn't let you withdraw sheets of material if the amount you are trying to withdraw. For example, if you are trying to withdraw 10 sheets worth of plasma but there is only enough for 8 sheets then clicking on the button to withdraw 10 sheets of plasma won't do anything. This is quite annoying when trying to withdraw large amounts of material and I believe that when you attempt to overdraw sheets, you should just completely empty the ORM of whatever material you're attempting to withdraw.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
closes #6956

CL: [Improvement] attempting to withdraw more sheets from the mineral silo and autolathes will dump all sheets they have now

### CODE REVIEW

File: UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs [hunk 1]
```diff
@@ -36,8 +40,9 @@ private void AddMaterial(ItemTrait material, int quantity)
 
 		private void ConsumeMaterial(ItemTrait material, int quantity)
 		{
-			MaterialList[material] -= quantity;
-			currentResources -= quantity;
+			var quantityToSubtract = (int) Mathf.Clamp(quantity, 0, Single.PositiveInfinity);
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: surely clamping to how much material you have would be better way of doing it?
so if you got 50 clamps maximum to 50?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: It's a safety precaution to make sure quantity never falls below 0. It's basically doing the same exact thing as before but making sure that quantity is never negative which the machine will try to do sometimes.

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: how does it become negative?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: 20 materials - 50 materials to consume (quantity) = -30 materials/resources

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: meant as in quantity , and surely you could just do 99999
 and then that would still equal
 MaterialList[material] = -99994

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: I reverted it because I can't replicate the issue anymore.

**Review by Bod9001 — DISMISSED** [review]

File: UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs [hunk 2]
```diff
@@ -56,11 +60,18 @@ public bool TryAddSheet(ItemTrait material, int quantity)
...
+				return quantity;
+			}
 			if (MaterialList[material] < Cm3PerSheet)
 			{
 				return 0;
 			}
-			var Cm3Used = Cm3PerSheet * quantity;
 			if (MaterialList[material] < Cm3Used)
 			{
 				quantity = MaterialList[material] / Cm3PerSheet;
```
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs:77]: Seems a little complex, and you're not poking the UI.

Wouldn't it be simpler instead to just remove the `if` condition on line `63` (before changes)? The method could be as little as five lines.

**Review by NoooneyDude — CHANGES_REQUESTED** [review]

File: UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs (hunk 2 above)
Comment (NoooneyDude) [inline UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: Seems a little complex: you're not poking the UI and you'll get more sheets out than is possible (e.g. 500 cm3 stored, player requests 50 sheets and gets 50 sheets even though there's only a quarter of a sheet stored).

Wouldn't it be simpler instead to just remove the `if` condition on line `63` (before changes) and use either the requested amount of sheets or the amount of available sheets, whichever is lower?
```suggestion
			quantity = Mathf.Min(quantity, MaterialList[material] / Cm3PerSheet);
```
Can't suggest changes on lines too far away so there's an exta `}`... @Github...

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: I don't understand what you want here?

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: Just do 

```
		public int TryRemoveSheet(ItemTrait material, int quantity)
		{
			quantity = Mathf.Min(quantity, MaterialList[material] / Cm3PerSheet);
			var Cm3Used = Cm3PerSheet * quantity;
			ConsumeMaterial(material, Cm3Used);
			UpdateGUIs.Invoke();
			return quantity;
		}
```

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Objects/Machines/MaterialStorage.cs]: Done and tested

**Comment by MaxIsJoe** [comment]
review pls

**Review by NoooneyDude — APPROVED** [review]

**Review by Bod9001 — APPROVED** [review]

**Pull request merged**